File: typescript/problem_61.ts
export function fastPow(x: number, y: number): number {
    if (y === 0) return 1;
    if (y < 0)   return 0;
    const half = fastPow(x, Math.floor(y / 2));
    return y % 2 === 0 ? half * half : half * half * x;
}

export function problem61(): void {
    console.log("Problem 61:");
    if (fastPow(2, 10) !== 1024) throw new Error("assertion failed");
    if (fastPow(3, 0)  !== 1)    throw new Error("assertion failed");
    if (fastPow(5, 3)  !== 125)  throw new Error("assertion failed");
    console.log("All Tests Passed");
}

File: rust/problem_61.rs
pub fn fast_pow(x: i64, y: i64) -> i64 {
    if y == 0 { return 1; }
    if y < 0  { return 0; }
    let half = fast_pow(x, y / 2);
    if y % 2 == 0 { half * half } else { half * half * x }
}

pub fn problem61() {
    println!("Problem 61:");
    assert_eq!(fast_pow(2, 10), 1024);
    assert_eq!(fast_pow(3, 0),  1);
    assert_eq!(fast_pow(5, 3),  125);
    println!("All Tests Passed");
}
